### PR TITLE
Support resource owner grant authenticator

### PR DESCRIPTION
**Pull request by unglaublicherdude**

### CODE REVIEW

File: .github/actions/vaas-scan-action/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# vaas-scan-action
+
+This is a github aciton with that one can scan changes in pull-requests with the G DATA product Verdict-as-a-Service (VaaS). VaaS is an Antivirus in the cloud, that can be utilized by a simple api.
```
Comment (pstadermann) [inline .github/actions/vaas-scan-action/README.md:3]:
```suggestion
This is a github action with that one can scan changes in pull-requests with the G DATA product Verdict-as-a-Service (VaaS). VaaS is an Antivirus in the cloud, that can be utilized by a simple api.
```

File: golang/vaas/cmd/git-scan/main.go [hunk 2]
```diff
@@ -30,14 +30,23 @@ func main() {
 	}
 	log.Println("targetBranch:", targetBranch)
 
-	clientID, exists := os.LookupEnv("VAAS_CLIENT_ID")
-	if !exists {
-		log.Fatal("no Client ID set")
+	clientID, clientIdExists := os.LookupEnv("VAAS_CLIENT_ID")
+	username, usernameExists := os.LookupEnv("VAAS_USERAME")
+	password, passwordExsits := os.LookupEnv("VAAS_PASSWORD")
```
Comment (pstadermann) [inline golang/vaas/cmd/git-scan/main.go:35]:
```suggestion
	password, passwordExists := os.LookupEnv("VAAS_PASSWORD")
```

File: golang/vaas/cmd/git-scan/main.go [hunk 3]
```diff
@@ -30,14 +30,23 @@ func main() {
...
+	clientSecret, clientSecretExists := os.LookupEnv("VAAS_CLIENT_SECRET")
+
+	if !clientIdExists {
+		log.Fatal("no client_id set")
+
 	}
-	clientSecret, exists := os.LookupEnv("VAAS_CLIENT_SECRET")
-	if !exists {
-		log.Fatal("no Client Secret set")
+	if clientIdExists && (!passwordExsits && !clientSecretExists) {
```
Comment (pstadermann) [inline golang/vaas/cmd/git-scan/main.go:42]:
```suggestion
	if clientIdExists && (!passwordExists && !clientSecretExists) {
```

File: golang/vaas/cmd/git-scan/main.go [hunk 4]
```diff
@@ -30,14 +30,23 @@ func main() {
...
+		log.Fatal("either password or client_secret must be set")
 	}
+
+	if usernameExists && (!passwordExsits || !clientIdExists) {
```
Comment (pstadermann) [inline golang/vaas/cmd/git-scan/main.go:46]:
```suggestion
	if usernameExists && (!passwordExists || !clientIdExists) {
```

**Review by pstadermann — CHANGES_REQUESTED** [review]

File: .github/actions/vaas-scan-action/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
+# vaas-scan-action
+
+This is a github action with that one can scan changes in pull-requests with the G DATA product Verdict-as-a-Service (VaaS). VaaS is an Antivirus in the cloud, that can be utilized by a simple api.
+
+## preconditions
+
+Before you can use this action, you need to aquire credentials. Please find more Information on the [Verdict-as-a-Service product page](https://www.gdata.de/business/security-services/verdict-as-a-service). If you want to test VaaS on your own, you can get credentials on our [registration page](https://vaas.gdata.de/login)
```
Comment (GermanCoding) [inline .github/actions/vaas-scan-action/README.md:7]: Shouldn't we link to the onboarding guide at https://vaas.gdata.de first? That guide contains the referral to `/login`.

File: golang/vaas/cmd/git-scan/main.go [hunk 6]
```diff
@@ -30,14 +30,23 @@ func main() {
...
+		log.Fatal("when using the username, the password and client_id must be set")
+	}
```
Comment (GermanCoding) [inline golang/vaas/cmd/git-scan/main.go]: We log fatal errors here, but we don't actually abort, right?

Comment (unglaublicherdude) [reply golang/vaas/cmd/git-scan/main.go]: Fatal does os.exit

https://pkg.go.dev/log#Fatal

File: .github/actions/vaas-scan-action/README.md (hunk 5 above)
Comment (unglaublicherdude) [reply .github/actions/vaas-scan-action/README.md:7]: yeah, had that in first, dunno, just gimme your suggestions

Comment (GermanCoding) [reply .github/actions/vaas-scan-action/README.md:7]: I would prefer the guide, it's makes it clearer what to do when I click on that link.

Comment (unglaublicherdude) [reply .github/actions/vaas-scan-action/README.md:7]: You are right. Will change that tomorrow, when nobody has a different opinion.

**Review by GermanCoding — APPROVED** [review]

**Pull request merged**